core/drivers/smsc/smsc91x.c

```diff
@@ -290,6 +295,59 @@ static inline void io_write32s(volatile uintptr_t addr, const void *buffer, int
 	}
 }
 
+
+
+TEE_Result send_data(u8 * data, size_t data_size)
+{
+	size_t data_len, rem_space;
+	struct sk_buff * skb = NULL;
+	u8* buffer_ptr;
+	if(data == NULL)
+	{
+		EMSG("%s: invalid data pointer passed", CARDNAME);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	skb = (struct sk_buff *)bstgw_ethpool_buf_alloc(NULL);
+
+	if(skb == NULL)
+	{
+		EMSG("%s: Failed to allocate buffer", CARDNAME);
+		return TEE_ERROR_OUT_OF_MEMORY;
+	}
+
+	/* Align IP header to 32 bits */
+	skb_reserve(skb, 2);
+
+	/*
+	 * If odd length: packet_len - 5,
+	 * otherwise packet_len - 6.
+	 * With the trailing ctrl byte it's packet_len - 4.
+	 */
+	data_len = data_size - 6;
```

Comment:
Comment and code do not match. Is it that we need to check the packet size and device to which it will be deducted ?